app/grandchallenge/components/views.py

```diff
@@ -288,3 +286,77 @@ def get_success_url(self):
 
     def get_success_message(self, cleaned_data):
         return f"{self.object._meta.verbose_name.title()} was successfully deleted"
+
+
+class BaseModelOptions(TextChoices):
+    ARCHIVE = Archive._meta.model_name
+    READER_STUDY = ReaderStudy._meta.model_name
+
+
+class CivSetListView(
+    LoginRequiredMixin, PermissionListMixin, PaginatedTableListView
+):
+    model = None
+    permission_required = None
+    raise_exception = True
+    template_name = "components/civ_set_list.html"
+    row_template = "components/civ_set_row.html"
+    search_fields = [
+        "pk",
+        "values__interface__title",
+        "values__image__name",
+        "values__file",
+    ]
+    extra_columns = []
+    text_align = "left"
+    default_sort_order = "asc"
+    base_object_lookup = None
+    extra_prefetch_fields = []
+    columns = [
+        Column(title="Values"),
+        Column(title="View"),
+        Column(title="Edit"),
+        Column(title="Remove"),
+    ]
+
+    def __init__(self):
+        if self.extra_columns:
+            self.columns = [*self.extra_columns, *self.columns]
+
+    @cached_property
+    def base_object(self):
+        return NotImplementedError
+
+    def render_row(self, *, object_, page_context):
+        return render_to_string(
+            self.row_template,
+            context={**page_context, "object": object_},
+        ).split("<split></split>")

```

Comment:
How does this differ from the implementation in `PaginatedTableListView`?

Reply:
Doesn't differ at all and can go, sorry, didn't realize the duplication there when creating this base view.